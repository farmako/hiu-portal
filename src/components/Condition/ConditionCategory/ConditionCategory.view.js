import React from 'react';
import * as PropTypes from 'prop-types';
import { getConceptDisplay } from '../../common/HealthInfo/FhirResourcesUtils';

const ConditionCategory = ({ condition }) => {
  if (condition.category) {
    const categories = [];
    condition.category.forEach((c) => {
      const category = getConceptDisplay(c);
      if (category) {
        categories.push(category);
      }
    });
    if (categories.length > 0) {
      return (
        <span>
          {' '}
          {`( ${
            categories.reduce((acc, value) => `${acc}, ${value}`)
          } )`}
          {' '}
        </span>
      );
    }
  }
  return null;
};

const conditionShape = PropTypes.shape({
  category: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
    coding: PropTypes.shape({
      code: PropTypes.string,
      display: PropTypes.string,
    }),
  })),
});

ConditionCategory.propTypes = {
  condition: PropTypes.shape(conditionShape),
};

ConditionCategory.defaultProps = {
  condition: {},
};

export default ConditionCategory;
