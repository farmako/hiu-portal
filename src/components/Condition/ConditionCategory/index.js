import ConditionCategory from './ConditionCategory.view';

export default ConditionCategory;
