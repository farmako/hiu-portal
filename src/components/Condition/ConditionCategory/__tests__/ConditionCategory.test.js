import React from 'react';
import { shallow } from 'enzyme';
import ConditionCategory from '../ConditionCategory.view';

describe('ConditionOnset', () => {
  const mockData = {
    category: [
      {
        text: 'Category 1',
        coding: {
          code: 'abc',
          display: 'abc text',
        },
      },
      {
        text: 'Category 2',
        coding: {
          code: 'abc',
          display: 'abc text',
        },
      },
    ],
  };
  const wrapper = shallow(
    <ConditionCategory condition={mockData} />
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render category text', () => {
    expect(wrapper.find('span').at(0).text()).toEqual(' ( Category 1, Category 2 ) ');
  });

  it('should return empty object if condition data is undefined', () => {
    wrapper.setProps({condition:{category:null}});
    expect(wrapper.find('span').exists()).toEqual(false);
  });
});
