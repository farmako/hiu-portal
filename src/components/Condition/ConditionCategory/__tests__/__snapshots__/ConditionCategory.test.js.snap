// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionOnset should render properly 1`] = `
<span>
   
  ( Category 1, Category 2 )
   
</span>
`;
